src/Chapter4.hs

```diff
@@ -488,7 +494,16 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+instance Applicative List where
+    pure a = Cons a Empty
 
+    Empty <*> _ = Empty
+    _ <*> Empty = Empty
+    Cons f fs <*> a = append (fmap f a) (fs <*> a)
+
+
+append Empty as = as
```

Comment:
In general, it's always a good idea to put top-level signatures for functions :slightly_smiling_face: